### DISCUSSION TITLE
Question related to NCCL + MPI

### DISCUSSION THREAD

**Discussion by ShivanshuPurohit** [mpi4py/mpi4py#25] [Q&A]
Does using `comm = MPI.COMM_WORLD` affect pytorch in _any_ way? Pytorch can be built on NCCL or MPI backends but supports only NCCL by default. I have seen situations where mpi4py works with pytorch seamlessly, but this is my question: Does the NCCL background of pytorch affect the use of mpi4py in any way? I'd like to know what happens under the hood when it works with NCCL built pytorch.

**Comment by dalcinl** [comment]
I would like to know, too. You should start looking at the [official warnings](https://docs.nvidia.com/deeplearning/nccl/user-guide/docs/mpi.html#inter-gpu-communication-with-cuda-aware-mpi).
